jtrust-lib/src/main/java/be/fedict/trust/crl/OnlineCrlRepository.java

```diff
@@ -87,50 +90,55 @@ public void setCredentials(final Credentials credentials) {
 	}
 
 	@Override
-	public X509CRL findCrl(final URI crlUri, final X509Certificate issuerCertificate,
-			final Date validationDate) {
+	public X509CRL findCrl(final URI crlUri, final X509Certificate issuerCertificate, final Date validationDate) throws ServerNotAvailableException {
 		try {
 			return getCrl(crlUri);
 		} catch (final CRLException e) {
 			LOG.debug("error parsing CRL: " + e.getMessage(), e);
 			return null;
-		} catch (final Exception e) {
+		} catch (final IOException | CertificateException | NoSuchProviderException | NoSuchParserException | StreamParsingException e) {
 			LOG.error("find CRL error: " + e.getMessage(), e);
 			return null;
 		}
 	}
 
 	private X509CRL getCrl(final URI crlUri) throws IOException,
 			CertificateException, CRLException, NoSuchProviderException,
-			NoSuchParserException, StreamParsingException {
+			NoSuchParserException, StreamParsingException, ServerNotAvailableException {
 		final DefaultHttpClient httpClient = new DefaultHttpClient();
+
 		if (null != this.networkConfig) {
-			final HttpHost proxy = new HttpHost(this.networkConfig.getProxyHost(),
-					this.networkConfig.getProxyPort());
-			httpClient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY,
-					proxy);
+			final HttpHost proxy = new HttpHost(this.networkConfig.getProxyHost(), this.networkConfig.getProxyPort());
+			httpClient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, proxy);
 		}
+
 		if (null != this.credentials) {
 			this.credentials.init(httpClient.getCredentialsProvider());
 		}
+
 		final String downloadUrl = crlUri.toURL().toString();
 		LOG.debug("downloading CRL from: " + downloadUrl);
 		final HttpGet httpGet = new HttpGet(downloadUrl);
 		httpGet.addHeader("User-Agent", "jTrust CRL Client");
+
 		final HttpResponse httpResponse = httpClient.execute(httpGet);
 		final StatusLine statusLine = httpResponse.getStatusLine();
 		final int statusCode = statusLine.getStatusCode();
+
+		if (statusCode >= HttpURLConnection.HTTP_INTERNAL_ERROR) {
+			LOG.error("CRL server responded with status code: " + statusCode );
```

Comment:
Remove logging, because you're already throwing an exception (which can be logged where you catch it).